### PR TITLE
add extra section-recipe maintainer and feedstock-name

### LINKED ISSUE #1769: document extras feedstock-name
We should document this feature for naming feedstocks differently than their package names in staged recipes:

```
extras:
   feedstock-name: <name>
```

I think this should go in the recipe section descriptions in the docs.

### PR DESCRIPTION
**Pull request by ssurbhi560**
Closes #1769 and #1331 

PR Checklist:

- [x] make all edits to the docs in the `src` directory, not in `docs` or in the html files
- [x] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [x] put any other relevant information below

### CODE REVIEW

**Review by beckermr — APPROVED** [review]

File: src/maintainer/adding_pkgs.rst [hunk 1]
```diff
@@ -761,6 +756,38 @@ For some languages, the community provides tools which can automate this process
...
+``@conda-forge-admin, please add user @username``
+
+where username is the GitHub username of the new maintainer to be added. Please refer to `Becoming a maintainer <https://conda-forge.org/docs/orga/guidelines.html#becoming-a-maintainer>`__ and `Updating the maintainer <https://conda-forge.org/docs/maintainer/updating_pkgs.html#updating-the-maintainer-list>`__ for detailed instructions.
+
+.. _feedstock_name:
+
+Feedstock name
+..............
+
+If you want the name of the feedstock different from the package name in the staged-recipes, you can use the ``feedstock-name`` directive in the recipe of that package, like this:
```
Comment (kathatherine) [inline src/maintainer/adding_pkgs.rst:782]: Let's put "to be" in front of "different"

**Review by kathatherine — CHANGES_REQUESTED** [review]
Just one thing! Otherwise, it looks great

**Review by kathatherine — APPROVED** [review]
LGTM!

**Comment by kathatherine** [comment]
@beckermr You good with merging this?

**Review by beckermr — APPROVED** [review]

**Comment by jakirkham** [comment]
This is great! Thank you 🙏

**Pull request merged**